test/fuzz/luaL_loadbuffer/serializer.cc

```diff
@@ -184,16 +184,99 @@ GetCondition(const std::string &counter_name, const std::string &then_block)
 	return retval;
 }
 
+/**
+ * Class that registers and provides context during code
+ * generation.
+ * Used to generate correct Lua code.
+ */
+class Context {
+public:
+	enum BlockType {
+		kReturnable,
+		kBreakable,
+	};
+
+	void step_in(BlockType type)
+	{
+		block_stack_.push(type);
+		if (type == BlockType::kReturnable) {
+			++returnable_counter_;
+		}
+	}
+
+	void step_out()
+	{
+		assert(!block_stack_.empty());
+		if (block_stack_.top() == BlockType::kReturnable) {
+			assert(returnable_counter_ > 0);
+			--returnable_counter_;
+		}
+		block_stack_.pop();
+	}
+
+	std::string get_next_block_setup()
+	{
+		std::size_t id = GetCounterIdProvider().next();
+		std::string counter_name = GetCounterName(id);
+
+		return GetCondition(counter_name, get_exit_statement_()) +
+		       GetCounterIncrement(counter_name);
+	}
+
+	bool break_is_possible()
+	{
+		return !block_stack_.empty() &&
+		       block_stack_.top() == BlockType::kBreakable;
+	}
+
+	bool return_is_possible()
+	{
+		return returnable_counter_ > 0;
+	}
+
+private:
+
+	std::string get_exit_statement_()
+	{
+		assert(!block_stack_.empty());
+		switch (block_stack_.top()) {
+		case BlockType::kBreakable:
+			return "break";
+		case BlockType::kReturnable:
+			return "return";
+		}
+	}
+
+	std::stack<BlockType> block_stack_;
+	/*
+	 * Returnable block can be exited with return from
```

Comment:
Typo: s/Returnable/The returnable/
Typo: s/Breakable/The breakable/

Reply:
Fixed.